### PR TITLE
Make "Setup Roster" step during classroom creation

### LINKED ISSUE #987: Rosters: Staffship 'make roster' step in classroom creation
When creating a classroom, part of the setup should be optionally setting up a roster.

This will probably be split across a few PRs.

TODO: Talk to @mozzadrella about copy.

mock:
screen shot 2017-06-27 at 10 17 01 am

### PR DESCRIPTION
**Pull request by nwoodthorpe**
Closes #987

This PR adds the "Setup Roster" step during classroom creation.

This is currently behind a feature flag.

In V1, the user pastes a list of the identifiers into a text area. V2 will allow for a csv upload, but this PR is already getting quite large.

The copy in this PR is a placeholder right now. @mozzadrella is helping with improved copy.

We also have a placeholder in one of the flash messages for when we add the roster management page. Since this is being staffshipped, it won't affect anyone.

screen shot 2017-07-05 at 4 06 15 pm

This is shown after the "invite other admins" step of classroom setup.

/cc @johndbritton
/review @tarebyte @wilfriedE

### CODE REVIEW

File: app/views/rosters/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    <div>
+      <%= form_tag roster_path(@organization), method: :post do |f| %>
+        <dl class="<%= view.form_class_for(:identifiers) %> short-field">
+          <dt><label>Paste your list of identifiers below (one per line, please!):</label></dt>
+          <dd><%= text_area_tag(:identifiers, '', class: 'form-control input-block') %></dd>
+          <%= render('shared/form_error', errors: view.error_message_for(:identifiers)) if view.errors_for?(:identifiers) %>
+        </dl>
+
+        <dl class="<%= view.form_class_for(:identifier_name) %> short-field">
+          <dt><label>Give this batch of identifiers a name (in case you'd like to re-use it):</label></dt>
```
Comment (nwoodthorpe) [inline app/views/rosters/new.html.erb]: This "name" isn't actually a unique identifier or a way to identify a roster. It's a human-readable description of what the identifiers are. E.g. "Student email addresses" or "UCSF student number"

@mozzadrella

File: config/routes.rb [hunk 2]
```diff
@@ -47,6 +47,8 @@
         patch :setup_organization
         get   'settings/invitations', to: 'organizations#invitation'
         get   'settings/teams',       to: 'organizations#show_groupings'
+
+        resource :roster
```
Comment (wilfriedE) [inline config/routes.rb]: This should limit to the accessible actions.

Comment (nwoodthorpe) [inline config/routes.rb]: Good catch, thanks!

File: app/views/rosters/new.html.erb (hunk 1 above)
Comment (mozzadrella) [inline app/views/rosters/new.html.erb]: On it!

**Comment by nwoodthorpe** [comment]
(Ignore all the Hound stuff, should be fixed now)

File: app/controllers/rosters_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  def create
+    @roster = Roster.new(identifier_name: params[:identifier_name])
+    @roster.save!
+
+    add_identifiers_to_roster
+
+    @organization.roster = @roster
+    @organization.save!
+
+    flash[:success] = 'Your classroom roster has been saved! Manage it HERE' # TODO: ADD LINK TO MANAGE PAGE
```
Comment (nwoodthorpe) [inline app/controllers/rosters_controller.rb]: The TODO here is addressed in the next PR (the manage feature doesn't exist in this commit). This is being staffshipped, so I'm pretty sure this is OK.

**Pull request merged**